content/5_Plat/HLD.mdx

```diff
@@ -15,6 +15,253 @@ frequency: 1
 
 ## Tutorial
 
+In some advanced problems on trees, the problem may ask you to support the
+following operations:
+
+- Update all nodes along the path from node $x$ to node $y$ (Note that this path
+  is unique due to properties of trees).
+
+- Find the sum, maximum, minimum (or any other operation that satisfies the
+  associative property) along the path from node $x$ to node $y$.
+
+Other simpler algorithms such as difference on trees or LCAs could be used to
+support any one of the operations. However, Heavy Light Decomposition (or HLD)
+is needed to support both.
+
+## Definitions:
+
+To understand HLD, I will first clarify the definition of these important terms:
+
+- A **heavy child** of a node is the child with the largest subtree size rooted
+  at the child.
+- A **light child** of a node is any child that is not a **heavy child**.
+- A **heavy edge** connects the current node to its **heavy child**.
+- A **light edge** connects the current node to any of its **light children**.
+- A **heavy path** is the path formed by a collection **heavy edges**.
+- A **light path** is the path formed by a collection **light edges**.
+
+## Properties:
+
+Given the definitions stated above, an important property can be derived (I will
+prove this a bit later).
+
+- Any path from node $x$ to node $y$ on the tree can pass through at most
+  $\mathcal{O}(\log N)$ light edges.
```

Comment:
I'd prefer you to move the proof here and put it in a spoiler so that you don't have to say "let's suppose the above property is true"